### PR TITLE
feat: use the new autogenerated openapi bindings for the rest server

### PR DESCRIPTION
**Pull request by tiagolobocastro**
feat: add opengenerated openapi rust bindings for the rest server

Add the pre-commit handler and the Jenkins check to make sure we don't forget to
regenerate the code when the spec is modified.

chore: add conversions to/from internal types to openapi

feat: use the new openapi server bindings

The rest server now configures itself using the new autogenerated openapi bindings.
In other other, we simply have to impl a trait for each resource type and put all
the logic there.

Currently these traits still have some actix type lingo, this was to make the transition
easier. It should be very simple to remove all actix types and have it become a simple
trait around results and Ser+DeSer traits.

Also add back the swagger ui using the static openapi specs file.

todo: use autogenerated code for the client side as well

### CODE REVIEW

File: scripts/update-openapi-generator [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
```
Comment (paulyoong) [inline scripts/update-openapi-generator:1]: The file is missing a ".sh" suffix.

**Review by paulyoong — APPROVED** [review]

File: control-plane/rest/service/src/v0/children.rs [hunk 2]
```diff
@@ -162,3 +100,75 @@ fn build_child_uri(child_id: ChildUri, req: HttpRequest) -> ChildUri {
         }
     })
 }
+
+#[async_trait::async_trait]
+impl apis::ChildrenApi for RestApi {
+    async fn del_nexus_child(
+        query: &str,
+        Path((nexus_id, child_id_)): Path<(String, String)>,
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply control-plane/rest/service/src/v0/children.rs:107]: Not that I know of, it's automatically added by the generator somehow when it escapes characters.

File: common/src/types/v0/message_bus/replica.rs [hunk 3]
```diff
@@ -97,6 +127,19 @@ impl ReplicaOwners {
...
 
+impl From<ReplicaOwners> for models::ReplicaSpecOwners {
+    fn from(src: ReplicaOwners) -> Self {
+        Self {
+            nexuses: src
+                .nexuses
+                .iter()
+                .map(|n| apis::Uuid::try_from(n).unwrap())
+                .collect(),
+            volume: src.volume.map(|n| apis::Uuid::try_from(n).unwrap()),
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply common/src/types/v0/message_bus/replica.rs:138]: As discussed, the ID types themselves will become UUID so we won't need the try_from at all

File: common/src/types/v0/message_bus/spec.rs [hunk 4]
```diff
@@ -21,3 +22,30 @@ pub struct Specs {
...
+//             from.iter().cloned().map(From::from).collect()
+//         }
+//         Self {
+//             nexuses: vec_to_vec(src.nexuses),
+//             pools: vec_to_vec(src.pools),
+//             replicas: vec_to_vec(src.replicas),
+//             volumes: vec_to_vec(src.volumes),
+//         }
+//     }
+// }
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply common/src/types/v0/message_bus/spec.rs:51]: dohh

File: scripts/update-openapi-generator (hunk 1 above)
Comment (tiagolobocastro) [reply scripts/update-openapi-generator:1]: dohhh

File: common/src/types/v0/message_bus/volume.rs [hunk 5]
```diff
@@ -53,10 +76,39 @@ impl From<(&VolumeId, &Nexus)> for Volume {
...
+}
+
 /// Volume State information
 /// Currently it's the same as the nexus
 pub type VolumeState = NexusState;
 
+impl From<VolumeState> for models::VolumeState {
+    fn from(src: VolumeState) -> Self {
+        match src {
+            VolumeState::Unknown => Self::Faulted,
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply common/src/types/v0/message_bus/volume.rs:95]: Yup, updated the VolumeState to have an unknown.

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**